docs/schema.graphql

```diff
@@ -1,1060 +0,0 @@
-  """
-  address: String!
-}
-
-enum AccountType {
-  META_CONTRACT
-  POLYJUICE_CONTRACT
-  POLYJUICE_ROOT
-  UDT
-  USER
-}
-
-type AccountUdt {
-  account: Account
-  address_hash: String
-  balance: Decimal
-  id: Int
-  inputs: String
-  udt: Udt
-}
-
-input AccountUdtContractAddressInput {
-  """
-  argument: the page of query result, the relations of postgres offset: offset = (page - 1) * page_size
-  
-  default: 1
-  """
-  page: Int = 1
-
-  """
-  argument: the page_size of query result, it's the same of postgres limit
-  
-  default: 20
-  """
-  page_size: Int = 20
-
-  """
-  argument: the sort of type by custom condition
-  
-  default: desc
-  """
-  sort_type: SortType = DESC
-  token_contract_address_hash: String!
-}
-
-input AccountUdtsInput {
-  """
-  argument: the list of account udt address
-  example: ["0x15ca4f2165ff0e798d9c7434010eaacc4d768d85"]
-  """
-  address_hashes: [String] = []
-
-  """
-  argument: the page of query result, the relations of postgres offset: offset = (page - 1) * page_size
-  
-  default: 1
-  """
-  page: Int = 1
-
-  """
-  argument: the page_size of query result, it's the same of postgres limit
-  
-  default: 20
-  """
-  page_size: Int = 20
-
-  """
-  argument: the address of smart contract which supply udts
-  example: "0xbf1f27daea43849b67f839fd101569daaa321e2c"
-  """
-  token_contract_address_hash: String
-}
-
-type Block {
-  account: Account
-  aggregator_id: Int
-  difficulty: Decimal
-  extra_data: String
-  gas_limit: Decimal
-  gas_used: Decimal
-  hash: String
-  layer1_block_number: Int
-  layer1_tx_hash: String
-  logs_bloom: String
-  nonce: String
-  number: Int
-  parent_hash: String
-  sha3_uncles: String
-  size: Int
-  state_root: String
-  status: BlockStatus
-  timestamp: DateTime
-  total_difficulty: Decimal
-  transaction_count: Int
-  transactions(input: PageAndSizeInput = {page: 1, page_size: 5}): [Transaction]
-}
-
-input BlockInput {
-  hash: String
-  number: Int
-}
-
-input BlocksInput {
-  """
-  argument: the page of query result, the relations of postgres offset: offset = (page - 1) * page_size
-  
-  default: 1
-  """
-  page: Int = 1
-
-  """
-  argument: the page_size of query result, it's the same of postgres limit
-  
-  default: 20
-  """
-  page_size: Int = 20
-
-  """
-  argument: the sort of type by custom condition
-  
-  default: desc
-  """
-  sort_type: SortType = DESC
-}
-
-enum BlockStatus {
-  COMMITTED
-  FINALIZED
-}
-
-"""
-The `DateTime` scalar type represents a date and time in the UTC
-timezone. The DateTime appears in a JSON response as an ISO8601 formatted
-string, including UTC timezone ("Z"). The parsed date and time string will
-be converted to UTC if there is an offset.
-"""
-scalar DateTime
-
-"""
-The `Decimal` scalar type represents signed double-precision fractional
-values parsed by the `Decimal` library.  The Decimal appears in a JSON
-response as a string to preserve precision.
-"""
-scalar Decimal
```

Comment:
Please provide some example of the JSON representation.

Is this `Decimal` type arbitrary precision? The doc lets me think it is a double float number, which is not suitable for storing balance.

Please pay attention to the precision issue. Add some test cases for common double float precision problems and check that the field is string in JSON.

Reply:
adding enrich description with this pr #395